src/shared/extensions.ts

```diff
@@ -1,9 +1,29 @@
 /*!
- * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
+import { AwsContext } from './awsContext'
+import { RegionProvider } from './regions/regionProvider'
+import { SettingsConfiguration } from './settingsConfiguration'
+import { TelemetryService } from './telemetry/telemetryService'
+import { ChannelLogger } from './utilities/vsCodeUtils'
+
 export const VSCODE_EXTENSION_ID = {
     awstoolkit: 'amazonwebservices.aws-toolkit-vscode',
     python: 'ms-python.python'
 }
+
+/**
+ * Long-lived, extension-scoped, shared globals.
+ */
+export interface ExtContext extends vscode.ExtensionContext {
+    //extensionContext: vscode.ExtensionContext
```

Comment:
> Why not have ExtContext contain a vscode.ExtensionContext instead of extend from one? This provides cleaner separation of objects/functionality being accessed from the object.

cleaner how? We are literally passing around a context. Having a context in a context is an inner platform.